plugins/out_prometheus_exporter/prom.c

```diff
@@ -175,44 +175,67 @@ static void cb_prom_flush(struct flb_event_chunk *event_chunk,
     int add_ts;
     size_t off = 0;
     flb_sds_t metrics;
-    cfl_sds_t text;
+    cfl_sds_t text = NULL;
+    cfl_sds_t tmp = NULL;
     struct cmt *cmt;
     struct prom_exporter *ctx = out_context;
+    int ok = CMT_DECODE_MSGPACK_SUCCESS;
+
+    text = flb_sds_create_size(128);
+    if (text == NULL) {
+        flb_plg_debug(ctx->ins, "failed to allocate buffer for text representation of metrics");
+        FLB_OUTPUT_RETURN(FLB_ERROR);
+    }
 
     /*
      * A new set of metrics has arrived, perform decoding, apply labels,
      * convert to Prometheus text format and store the output in the
      * hash table for metrics.
+     * Note that metrics might be concatenated. So, we need to consume
+     * until the end of event_chunk.
      */
-    ret = cmt_decode_msgpack_create(&cmt,
-                                    (char *) event_chunk->data,
-                                    event_chunk->size, &off);
-    if (ret != 0) {
-        FLB_OUTPUT_RETURN(FLB_ERROR);
-    }
+    while ((ret = cmt_decode_msgpack_create(&cmt,
+                                            (char *) event_chunk->data,
+                                            event_chunk->size, &off)) == ok) {
 
-    /* append labels set by config */
-    append_labels(ctx, cmt);
+        if (ret != 0) {
+            FLB_OUTPUT_RETURN(FLB_ERROR);
```

Comment:
You need to release `text` here

Reply:
Thanks. I fixed it in https://github.com/fluent/fluent-bit/pull/9122/commits/8ada2be6c46eb68f4a9d432a8f8f198bab07a35b.